### ISSUE TITLE
Multi-Select can't be exited

### ISSUE THREAD

**Issue by jeshyr** [glyphr-studio/Glyphr-Studio-1#254]
Minimal setup to reproduce - I tested this on both Safari and Firefox for Mac.

Open Modegg example project
Click on one of the four points on the crossbar of the "A"
Hold down command key and click on another of the four points - note that the "+" glyph shows on the cursor as you do this, indicating multi-select mode.
Release command key - this should get you back to single select-mode but doesn't. "+" glyph stays on the cursor, you can't drag a handle or a single node, etc.

It appears that after you multi-select something you are stuck in multi select mode forever. I have tried option-click and shift-click and command-click and control-click and right click and escape and occasionally something kicks it back to single-select mode but I couldn't quite figure out what. Nothing seemed to work repeatably except saving the file and reloading the browser, which of course is not so practical.

**Comment by mattlag** [comment]
Interesting - this works on Windows Chrome / Firefox.  I'll look into why this may be happening on Mac, see if there is something I can do.  

If you figure out a specific key that always allows you to exit multi-select mode, let me know.

**Comment by jeshyr** [comment]
I tried this today in Google Chrome for Mac just for completeness and the same behaviour is evident. It looks like the OS isn't properly exposing the "key up" for the command key?

I did realise on Google Chrome that if I use the control key and control click two points it gets me out of multi-select mode but since control-click also brings up the context menu (aka right click menu) it's very awkward to do -you have to control-click one point and then find another point that's not obscured by the context menu and (still holding down control) click that and it's only the *second* click that's getting back to single-select mode. I suspect the first one is somehow eaten by the OS because it brings up the context menu? 

... but then I tried the exact same thing again and it didn't work. Third try worked, fourth try didn't. Either this is random or it's mediated by something I can't figure out. Feel like I'm going crazy!!

**Comment by jeshyr** [comment]
I think I figured out what's going on. Mac OS X reports "command" key as either keycode 91 (left) or 93 (right). framework_events_keyboard.js getKeyFromEvent reports this is a "meta" key.

Your code at /framework_events_keyboard.js line 44 is fine with this - it accepts ctrl or meta to enter multi select mode.

But the keyup() function at line 8 is only looking for ctrl and will *not* accept meta.

That still doesn't explain why ctrl is only working intermittently to exit multi select mode... but it's half the problem at least.

**Comment by mattlag** [comment]
Awesome, thanks for looking into this.

**Comment by mattlag** [comment]
Hi @jeshyr 
This will be fixed in v1.08 (to be released soon).  I made the update like you requested, adding the support for both meta keys to the keyup event.

I also added an explicit 'exit multi select mode' functionality: if you have the 'shape select' arrow selected, and enter multi-select mode but can't exit, clicking the 'shape select' arrow *again* will exit multi select mode.  Same for the pen tool selecting multiple path points - selecting the pen tool again will exit multi-select mode.
